runtime/gc_vlhgc/VLHGCAccessBarrier.cpp

```diff
@@ -281,12 +284,47 @@ MM_VLHGCAccessBarrier::jniGetPrimitiveArrayCritical(J9VMThread* vmThread, jarray
 		*isCopy = JNI_FALSE;
 	}
 
-	if (alwaysCopyInCritical || !indexableObjectModel->isInlineContiguousArraylet(arrayObject)) {
+	if (alwaysCopyInCritical) {
```

Comment:
To make eventual removal of double mapping easier we could do this (but don't do it yet)

```suggestion
#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
	if (alwaysCopyInCritical) {
#else /* defined(J9VM_GC_ENABLE_DOUBLE_MAP) */
	if (alwaysCopyInCritical || !indexableObjectModel->isInlineContiguousArraylet(arrayObject)) {
#endif /* defined(J9VM_GC_ENABLE_DOUBLE_MAP) */
```